src/shims/time.rs

```diff
@@ -1,8 +1,71 @@
+use std::sync::atomic::AtomicU64;
+use std::sync::Arc;
 use std::time::{Duration, Instant, SystemTime};
 
+use rustc_data_structures::sync::Ordering;
+
 use crate::concurrency::thread::Time;
 use crate::*;
 
+#[derive(Debug)]
+pub enum TimeAnchor {
+    Host(Instant),
+    Virtual(Arc<AtomicU64>),
+}
+
+impl TimeAnchor {
+    pub fn new(communicate: bool) -> Self {
+        if communicate { Self::Host(Instant::now()) } else { Self::Virtual(Arc::new(0.into())) }
+    }
+
+    pub fn get(&self) -> Duration {
+        match self {
+            Self::Host(instant) => Instant::now().saturating_duration_since(*instant),
+            Self::Virtual(nanoseconds) => Duration::from_nanos(nanoseconds.load(Ordering::Relaxed)),
+        }
+    }
+
+    pub fn tick(&self) {
+        match self {
+            Self::Host(_) => (),
+            Self::Virtual(nanoseconds) => {
+                nanoseconds.fetch_add(1, Ordering::Relaxed);
```

Comment:
One nanosecond per basic block? Wow that's a fast machine.^^

Please move this into a constant somewhere, like `NANOSECOND_PER_BASIC_BLOCK` or so, so that it is clear what this means. Also maybe 10ns would be a bit more realistic? Not that it really matters...

Reply:
> One nanosecond per basic block? Wow that's a fast machine.^^

You wanted miri to be the fastest, didn't you?

> Please move this into a constant somewhere, like `NANOSECOND_PER_BASIC_BLOCK` or so, so that it is clear what this means. Also maybe 10ns would be a bit more realistic? Not that it really matters...

Sure I'll do the change